### PR TITLE
Single Page Content Build Lambda Using Preview Server

### PR DESCRIPTION
**Pull request by indytechcook**
## Description

This is a POC for generating publishing a single page using the preview server to generate HTML.  This will align with a devops PR which is still work in progress.

https://github.com/department-of-veterans-affairs/va.gov-cms/issues/5429

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/lambdas/single-page-build.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+exports.handler = async function(event, context) {
+  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+  console.log(`Context: ${JSON.stringify(context, null, 2)}`);
+
+  for (const record of event.Records) {
+    const { body } = record;
+    if (body.nid && body.path) {
+      // eslint-disable-next-line no-await-in-loop
+      await processSinglePage(body.nid, body.path);
```
Comment (acrollet) [inline src/platform/lambdas/single-page-build.js:44]: perhaps use e.g. `await Promise.all(pageArray)` after the loop?

File: src/platform/lambdas/single-page-build.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-console */
+
+const s3 = require('aws-sdk/client/c3'); // eslint-disable-line import/no-unresolved
```
Comment (acrollet) [inline src/platform/lambdas/single-page-build.js:3]:
```suggestion
const s3 = require('aws-sdk/client/s3'); // eslint-disable-line import/no-unresolved
```

?

**Comment by indytechcook** [comment]
This PR requires https://github.com/department-of-veterans-affairs/content-build/pull/163 to be merged first

**Comment by indytechcook** [comment]
The lambda works now.

image

Here is the test event:

```
{
  "Records": [
    {
      "messageId": "19dd0b57-b21e-4ac1-bd88-01bbb068cb78",
      "receiptHandle": "MessageReceiptHandle",
      "body": {
        "nid": "782",
        "path": "pittsburgh-health-care/ambulatory-care-center-floor-plans"
      },
      "attributes": {
        "ApproximateReceiveCount": "1",
        "SentTimestamp": "1523232000000",
        "SenderId": "123456789012",
        "ApproximateFirstReceiveTimestamp": "1523232000001"
      },
      "messageAttributes": {},
      "md5OfBody": "{{{md5_of_body}}}",
      "eventSource": "aws:sqs",
      "eventSourceARN": "arn:aws-us-gov:sqs:us-gov-west-1:123456789012:MyQueue",
      "awsRegion": "us-gov-west-1"
    }
  ]
}
```

File: src/platform/lambdas/single-page-build.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ * @param context
+ */
+exports.handler = async function(event, context) {
+  console.log(`Event: ${JSON.stringify(event, null, 2)}`);
+  console.log(`Context: ${JSON.stringify(context, null, 2)}`);
+
+  const processPromises = [];
+  for (const record of event.Records) {
+    const { body } = record;
+    if (body.nid && body.path) {
```
[the remark this answers is not in the document]
Comment (indytechcook) [reply src/platform/lambdas/single-page-build.js:51]: No, node appears to do this on it's own.

**Pull request closed without merging**